packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -47,14 +88,25 @@ async function mockSearchResponse( items ) {
 		type: 'post',
 		url: `https://this/is/a/test/search/${ slug }`,
 	} ) );
-
 	await setUpResponseMocking( [
 		{
 			match: ( request ) =>
 				request.url().includes( `rest_route` ) &&
 				request.url().includes( `search` ),
 			onRequestMatch: createJSONResponse( mappedItems ),
 		},
+		...getPagesMocks( {
+			GET: [
+				{
+					type: 'page',
+					id: 1,
+					link: 'https://example.com/1',
+					title: {
+						rendered: 'My page',
+					},
+				},
+			],
+		} ),
```

Comment:
What's the purpose of these changes?

Reply:
When checking for the validity of the entity the navigation item links to, we have to call `getEntityRecord`, which throws a 404 error when it doesn't find an entity with the given parameters.

In this test suite, two tests are using the search box mocking only the search results, but not the underlying entities, which was causing `getEntityRecord` to throw the error, which would subsequently make the tests fail. There's a `createPages` function within this suite, but it didn't solve the problem, as mocked search results don't seem to care about entities that actually exist in the wordpress instance used for testing. Mocking the linked entities solved this problem.